MISRA.md

```diff
@@ -0,0 +1,113 @@
+# MISRA Compliance
+
+The FreeRTOS-Cellular-Interface library files conform to the [MISRA C:2012](https://www.misra.org.uk/MISRAHome/MISRAC2012/tabid/196/Default.aspx)
+guidelines, with the deviations listed below. Compliance is checked with Coverity static analysis.
+The specific deviations, suppressed inline, are listed below.
+
+Additionally, [MISRA configuration file](https://github.com/FreeRTOS/FreeRTOS-Cellular-Interface/blob/main/tools/coverity/misra.config) contains the project wide deviations.
+
+### Suppressed with Coverity Comments
+To find the violation references in the source files run grep on the source code
+with ( Assuming rule 10.4 violation; with justification in point 1 ):
+```
+grep 'MISRA Ref 10.4.1' . -rI
+```
+
+#### Directive 4.6
+_Ref 4.6.1_
+
+- MISRA C-2012 Directive 4.6 warns against using types that do not contain size
+    and sign information. The use of isalpha, isdigit, and isspace set this directive
+    off since they use base types that don't contain this information. We do not have
+    control over these functions so we are suppressing these violations.
+
+#### Directive 4.7
+_Ref 4.7.1_
+
+- MISRA C 2012 Directive 4.7 requires that when a function returns error information,
+    that error information shall be tested. The following line violates MISRA rule 4.7
+    because return value of strtol() is not checked for error. This violation is
+    justified because error checking by "errno" for any POSIX API is not thread
+    safe in FreeRTOS unless "configUSE_POSIX_ERRNO" is enabled. In order to avoid the
+    dependency on this feature, errno variable is not used.
+
+#### Rule 10.4
+_Ref 10.4.1_
+
+- MISRA C-2012 Rule 10.4 warns about using different types in an operation.
+    This rule is being flagged because of use of the standard library functions
+    isalpha, isdigit, and isspace. We do not have control over these functions so we
+    are suppressing these violations.
+
+#### Rule 10.5
+_Ref 10.5.1_
+
+- MISRA C-2012 Rule 10.5 Converting to an enum type. The variables
+    are checked to ensure that they are valid, and within a valid range.
+    Hence, assigning the value of the variable with a enum cast.
+
+#### Rule 10.8
+_Ref 10.8.1_
+
+- MISRA C-2012 Rule 10.8 warns about casting from unsigned to signed types.
+    This rule is being flagged because of use of the standard library functions
+    isalpha and isdigit. We do not have control over these so we are suppressing
+    these violations.
+
+#### Rule 11.3
+_Ref 11.3.1_
```

Comment:
I could not find this reference in the diff. Am I missing something?

Reply:
Yeah there are 3 occurrences of it in cellular_common_api.c 
[One](https://github.com/FreeRTOS/FreeRTOS-Cellular-Interface/pull/102/files#diff-03d83bb57d321395beb9eaf6f38c54175305aa652a173ef546bc7ef16e1a58aeR75), [Two](https://github.com/FreeRTOS/FreeRTOS-Cellular-Interface/pull/102/files#diff-03d83bb57d321395beb9eaf6f38c54175305aa652a173ef546bc7ef16e1a58aeR90), [Three](https://github.com/FreeRTOS/FreeRTOS-Cellular-Interface/pull/102/files#diff-03d83bb57d321395beb9eaf6f38c54175305aa652a173ef546bc7ef16e1a58aeR119)
I had tried a few things, but I felt that because we were passing in a variable that was being used to ensure that these are valid casts that suppressing these was fine.